flow/flat_buffers.cpp

```diff
@@ -452,7 +452,11 @@ TEST_CASE("/flow/FlatBuffers/VectorRef") {
 			serializedVector = StringRef(readerArena, writer.toStringRef());
 		}
 		ArenaObjectReader reader(readerArena, serializedVector, Unversioned());
-		reader.deserialize(FileIdentifierFor<decltype(outVec)>::value, vecArena, outVec);
+		// The VectorRef and Arena arguments are intentionally in a different order from the serialize call above.
+		// Arenas need to get serialized after any Ref types whose memory they own. In order for schema evolution to be
```

Comment:
I'll go ahead and merge this PR, but as a parting thought I'm curious whether there's any way we could enforce this or detect misuses outside of tools like valgrind.